### ISSUE TITLE
Data preprocess

### ISSUE THREAD

**Issue by Anmisdwx** [Raiiyf/DiffuSETS_Exp#2]
Hi, thanks for the great work!
Could you please share how to preprocess the data before training?  Any guidance or example script would be appreciated.

Thanks!

**Comment by Raiiyf** [comment]
Thank you for your feedback. 

We only perform signal resampling to original ECG waveform (5000 -> 1024) for the training of VAE, which converts all the ECG signals into latents afterwards. These latents are used for the training of latent diffusion model, without further processing.

For the implement details and  process of other labels, please refer to the dataset files under `dataset` and the `vae_encoding.py` under `vae`.

**Comment by Anmisdwx** [comment]
Thanks for your detailed reply!

I noticed in your repo there is a file '/data/0shared/laiyongfan/data_text2ecg/exclude_list.txt' mentioned, I tried to follow the preprocessing steps described in your paper, but only excluded about 700 samples.  Could you please help clarify or provide the exclude_list file?

Thanks a lot for your help!

**Comment by Raiiyf** [comment]
This is the exclude_list.txt file, hope it can help you.

[exclude_list.txt](https://github.com/user-attachments/files/21017306/exclude_list.txt)

**Comment by Anmisdwx** [comment]
Thanks a lot!

**Comment by Anmisdwx** [comment]
Sorry to bother you again!

I have a quick question about the training/testing split in DiffuSETS_train.py. From reading the code, it seems that all the data in mimic_vae_0_new.pt is used for training — is that correct?

Also, during evaluation, is the file mimic_vae_lite_0_new.pt (which contains 10,000 samples) used as the test set to compute evaluation metrics?

**Comment by Raiiyf** [comment]
Yes.

**Comment by Anmisdwx** [comment]
I noticed that mimic_vae_0_new.pt contains all 794,372 samples, including 10,000 test samples. Does this mean the test data is used during training? Won't this affect the metrics evaluation?

or is mimic_vae_0_new.pt different from the training dataset specified in the config file"./prerequisites/mimic_vae_lite.pt（not found in huggingface）?

**Comment by Raiiyf** [comment]
Good questions! Yes, and I understand your concerns about data leakage.

However, in our opinion, the effect of data leakage is minute in our tasks. During the generation process, the diffusion model has to sample noise from Gaussian distribution for result diversity, therefore the model cannot cheat by memorizing 'correct answers'. Moreover, the test dataset is built mainly for getting genuine input conditions and since the data leakage can be ignore, we do not separate the data to make the train dataset as large as possible.
